// @aws-sdk/client-cognito-identity/dist-es/commands/GetPrincipalTagAttributeMapCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetPrincipalTagAttributeMapInput, GetPrincipalTagAttributeMapResponse } from "../models/models_0";
import { deserializeAws_json1_1GetPrincipalTagAttributeMapCommand, serializeAws_json1_1GetPrincipalTagAttributeMapCommand, } from "../protocols/Aws_json1_1";
var GetPrincipalTagAttributeMapCommand = (function (_super) {
    __extends(GetPrincipalTagAttributeMapCommand, _super);
    function GetPrincipalTagAttributeMapCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetPrincipalTagAttributeMapCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityClient";
        var commandName = "GetPrincipalTagAttributeMapCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetPrincipalTagAttributeMapInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetPrincipalTagAttributeMapResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetPrincipalTagAttributeMapCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetPrincipalTagAttributeMapCommand(input, context);
    };
    GetPrincipalTagAttributeMapCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetPrincipalTagAttributeMapCommand(output, context);
    };
    return GetPrincipalTagAttributeMapCommand;
}($Command));
export { GetPrincipalTagAttributeMapCommand };
